mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -381,15 +396,28 @@ void CHudMapFinishedDialog::SetVisible(bool bVisible)
 
 void CHudMapFinishedDialog::SetCurrentPage(int pageNum)
 {
+    if (!engine->IsInGame())
+        return;
+
+    if (m_bPbNeedUpdate)
+    {
+        if (LoadPlayerBestTime())
+            m_bPbNeedUpdate = false;
+    }
```

Comment:
Let's flip this logic. I'd like to remove `m_fOverallDiff` in favor of just setting `m_bPbNeedsUpdate` inside the `replay_saved` event handler, then when this panel is closed, running this block of code I highlighted.